src/sources/internal_metrics.rs

```diff
@@ -75,14 +75,16 @@ impl SourceConfig for InternalMetricsConfig {
             if tag.is_empty() {
                 None
             } else {
-                Some(log_schema().host_key())
+                Some(tag.to_owned())
+            }
```

Comment:
Note that this can be phrased as:
```rust
(!tag.is_empty()).then(|| tag.to_owned())
```
(also below)

Reply:
Ooh, yes, I like that.